### PR TITLE
Add option to run on top of Redis

### PR DESCRIPTION
**Pull request by gregturn**
* Make entire S3 module optional
* Switch Cassandra to be disabled by default

### CODE REVIEW

**Comment by tomaslin** [comment]
the bulk of these changes should be in https://github.com/spinnaker/scheduled-actions/

**Comment by gregturn** [comment]
I thought that library was for echo.

**Comment by tomaslin** [comment]
yeah, sorry nvm

**Comment by gregturn** [comment]
Rebased and implemented the new history APIs.

@spinnaker/netflix-reviewers PTAL

File: front50-cassandra/src/main/groovy/com/netflix/spinnaker/front50/config/CassandraConfig.groovy [hunk 1]
```diff
@@ -28,8 +28,9 @@ import org.springframework.context.annotation.Configuration
 import org.springframework.core.env.Environment
 
 @Configuration
-@ConditionalOnExpression('${cassandra.enabled:true}')
+@ConditionalOnExpression('${spinnaker.cassandra.enabled:false}')
```
Comment (ajordens) [inline front50-cassandra/src/main/groovy/com/netflix/spinnaker/front50/config/CassandraConfig.groovy]: I believe we were using `cassandra.enabled` previously as there are some `kork-cassandra` components that use that key to enable/disable themselves.

File: front50-redis/front50-redis.gradle [hunk 2]
```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2014 Netflix, Inc.
```
Comment (ajordens) [inline front50-redis/front50-redis.gradle]: Copyright doesn't seem quite right

File: front50-redis/src/main/groovy/com/netflix/spinnaker/front50/redis/RedisApplicationDAO.groovy [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+import com.netflix.spinnaker.front50.exception.NotFoundException
+import com.netflix.spinnaker.front50.model.application.Application
+import com.netflix.spinnaker.front50.model.application.ApplicationDAO
+import org.springframework.data.redis.core.RedisTemplate
+import org.springframework.data.redis.core.ScanOptions
+/**
+ * Implementation of {@link ApplicationDAO} interface, leveraging {@link RedisTemplate} to do the
+ * heavy lifting.
+ *
+ * @author Greg Turnquist
```
Comment (ajordens) [inline front50-redis/src/main/groovy/com/netflix/spinnaker/front50/redis/RedisApplicationDAO.groovy]: No `@author` tags please

**Comment by gregturn** [comment]
. Addressed license headers
. Stripped out @author tags
. Restored cassandra.enabled

**Comment by ajordens** [comment]
Ze builds failed on this one

**Comment by gregturn** [comment]
@ajordens I forgot to re-test EVERYTHING. My tweak to the cassandra impacted its unit test. That has been resolved and verified locally. Let's see how this plays out on travis.

**Comment by ajordens** [comment]
LGTM

Can you add something to http://www.spinnaker.io/docs around migrating to this (what config needs to change, etc.)

**Comment by gregturn** [comment]
@ajordens I've created http://www.spinnaker.io/docs/front50-cassandra-to-redis

**Pull request merged**